### ISSUE TITLE
npe in ClientInvocationHandler

### ISSUE THREAD

**Issue by philippeu** [restlet/restlet-framework-java#1004]
ClientInvocationHandler is expecting a Status annotation and will throw NPE if not present.

```java
for (Class<?> clazz : javaMethod.getExceptionTypes()) {
    ThrowableAnnotationInfo tai = getThrowableAnnotationInfo(clazz);
    // tai could be null here
    if (tai.getStatus().getCode() == errorCode) {
        return tai;
    }
}
```
in addition, if the annotation is there, it is expecting the response is StatusInfo

```java
// explicit deserialization to StatusInfo
StatusInfo si = getClientResource().toObject(response.getEntity(), StatusInfo.class);
```

**Comment by thboileau** [comment]
Hello,
thanks for reporting the NPE, I've fixed that in 2.3 and master branches.
Regarding the second issue, I'm not sure  this is a problem, since the code checks before that the ThrowableAnnotationInfo is serializable or not. Do you have any test case in mind?

**Comment by philippeu** [comment]
Hi,
Thanks for the quick fix!

For the second point, if tai == null, we go to that part of code

```java
else if (response.isEntityAvailable()) {
   StatusInfo si = getClientResource().toObject(response.getEntity(), StatusInfo.class);

```

in my case, i don't have a Status annotation and i got a deserialization exception.

**Comment by thboileau** [comment]
Hi Philippe,

something is not clear for me. I guess you have an annotated method in a service such as:
```java
@Get
public String hello() throws MyException;
```

In your case, MyException inherits from the Exception class, and is doesn't contain any `@Status` annotation.
When the MyException is thrown, and because it is not annotated, the status service converts an instance of StatusInfo.
By doing so, on client side, you should really have an instance of StatusInfo to deserialize. Am I wrong?

**Comment by philippeu** [comment]
Hi Thierry,
You are correct, after checking more carefully the code, I am overwritting getRepresentation() in StatusService, then I got this exception:

```
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "errors" (class org.restlet.engine.application.StatusInfo), not marked as ignorable (6 known properties: , "homeRef", "contactEmail", "uri", "description", "code", "reasonPhrase"])
 at [Source: java.io.PushbackInputStream@40aa66fd; line: 1, column: 12] (through reference chain: org.restlet.engine.application.StatusInfo["errors"])
```

**Comment by thboileau** [comment]
Hi @philippeu , another issue related to yours. I've added a comment.
I expection a decision quite quickly.

**Comment by thboileau** [comment]
Hi @philippeu, we've removed the automatic deserialization of StatusInfo bean.
The fix will be provided by RF 2.3.1 coming very soon.

**Comment by philippeu** [comment]
Thanks Thierry
